post_office/tasks.py

```diff
@@ -5,28 +5,32 @@
 from post_office.mail import send_queued
 from post_office.utils import cleanup_expired_mails
 
+from .settings import get_celery_enabled
+
 # Only define the tasks and handler if we can import celery.
 # This allows the module to be imported in environments without Celery, for
 # example by other task queue systems such as Huey, which use the same pattern
 # of auto-discovering tasks in "tasks" submodules
-try:
-    from celery import shared_task
-except ImportError:
-    pass
-else:
-    @shared_task(ignore_result=True)
-    def send_queued_mail(*args, **kwargs):
-        send_queued()
 
-    def queued_mail_handler(sender, **kwargs):
-        """
-        To be called by post_office.signals.email_queued.send()
-        """
-        send_queued_mail.delay()
+if get_celery_enabled():
```

Comment:
This can be refactored into the `try`-block, without needing another level of indentation.

Reply:
Where do you want tu pot it ? We can't put an elif after an except. We will have a indentation level even if I put it into the else

My logic is if celery is not enabled  we do nothings even the import statement.

Reply:
this would keep the current indentation level:

```python
try:
    if get_celery_enabled():
        from celery import shared_task
    else:
        raise NotImplementedError()
except (ImportError, NotImplementedError):
    pass
else:
    ...
```